src/js/components/DataTable/stories/SpaceX.js

```diff
@@ -0,0 +1,196 @@
+  }
+  const result = count ? items.slice(start, start + count) : items;
+  return result;
+};
+
+export const SpaceX = () => {
+  const [groups, setGroups] = useState([]);
+  const [expanded, setExpanded] = useState([]);
+  const [sort, setSort] = useState({ property: 'name', direction: 'asc'});
+  const [data, setData] = useState([]);
+  const [limit, setLimit] = useState(20);
+
+  const expandable = useMemo(() => groups.map(({id}) => id), [groups]);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      const query = {
+        options: {
+          select: [ "name" ],
+        },
+      };
+      fetch("https://api.spacexdata.com/v4/rockets/query", {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(query),
+      })
+      .then(response => response.json())
+      .then(d => {
+        setGroups(d.docs);
+      })
+      .catch(error => console.error('Unable to get groups:', error));
+    };
+
+    fetchData();
+  }, []);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      if (groups.length === 0 || expanded.length === 0) {
+        setData(processData({
+          expanded,
+          show: 0,
+          count: limit,
+        }, groups, []));
+        return;
+      }
+      const query = {
+        options: {
+          populate: [
+            {
+              path: "rocket",
+              select: { name : 1},
+            },
+          ],
+          sort: {
+            rocket: "asc",
+            [sort.property || "name"]: sort.direction || "asc",
+          },
+          select: [ "name", "success", "failures" ],
+          limit,
+        },
+        query: {
+          rocket: {
+            "$in": expanded,
+          },
+        },
+      };
+      fetch("https://api.spacexdata.com/v4/launches/query", {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(query),
+      })
+      .then(response => response.json())
+      .then(d => {
+        setData(processData({
+          expanded,
+          show: 0,
+          count: limit,
+        }, groups, d.docs));
+      })
+      .catch(error => console.error('Unable to get data:', error));
+    };
+
+    fetchData();
+  }, [expanded, groups, limit, sort]);
+
+  return (
+    <Grommet theme={grommet}>
+      <Box align="center" pad="large">
+        <DataTable
+          primaryKey="id"
+          columns={columns}
+          data={data}
+          sortable
+          replace
+          groupBy={{
+            expandable,
+            expand: expanded,
+            property: 'rocketId',
+          }}
+          onUpdate={opts => {
+            setExpanded(opts.expanded);
+            setLimit(opts.count);
+            if (opts.sort) setSort(opts.sort);
+          }}
+          step={20}
+        />
+      </Box>
+    </Grommet>
+  );
+};
+
+SpaceX.storyName = 'SpaceX Grouped';
```

Comment:
I think we should disable this story from Chromatic since the data can change as more rockets are launched.